### ISSUE TITLE
Inconsistent behaviour in navigation on a line containing Japanese symbols

### ISSUE THREAD

**Issue by ProgerXP** [ProgerXP/Notepad2e#247]
This must be Scintilla's issue because Notepad++ has the same problem. See if this can be localized and reported to them.

Normally, if a line wraps (line wrapping must be enabled - Ctrl+W) pressing End puts the caret to the end of the current subline or (next End) - to the end of the line. So no matter how many sublines a line is wrapped to, pressing End twice will put the caret to the end.

Same with Home and with Shift+Home/End.

However, if a line consists of Japanese symbols - Home/End navigate to the preceding/following subline. So a long wrapped line will need Home/End pressed multiple times, not just 2. Is this intentional and if so then why/what for?

For tests you can create a long line by copy/pasting the following two symbols: `日本`

**Comment by cshnik** [comment]
Fixed.

**Comment by ProgerXP** [comment]
This is a Scintilla bug? Provide details please.

**Comment by cshnik** [comment]
> This is a Scintilla bug? Provide details please.

Yes, there was a problem with cursor position calculation when working with unicode text.
Updated Scintilla_ChangeLog.md.

**Comment by ProgerXP** [comment]
I mean if it's a Scintilla bug that should be reported to them or it's a problem with our patches?

**Comment by cshnik** [comment]
> I mean if it's a Scintilla bug that should be reported to them or it's a problem with our patches?

Created bug for Scintilla: [#2231 Cursor navigation issue in word wrap mode](https://sourceforge.net/p/scintilla/bugs/2231/)
